src/reporters/json.ts

```diff
@@ -27,18 +27,119 @@ import BaseReporter from './base';
 import { formatError, getTimestamp } from '../helpers';
 import { Journey, Step } from '../dsl';
 import snakeCaseKeys from 'snakecase-keys';
+import { NetworkInfo } from '../common_types';
 
 /* eslint-disable @typescript-eslint/no-var-requires */
-const programVersion = require('../../package.json').version;
+const { version, name } = require('../../package.json');
+
+type OutputType =
+  | 'journey/register'
+  | 'journey/start'
+  | 'step/screenshot'
+  | 'step/end'
+  | 'journey/network_info'
+  | 'journey/filmstrips'
+  | 'journey/browserconsole'
+  | 'journey/end';
+
+type OutputFields = {
+  type: OutputType;
+  journey: Journey;
+  timestamp?: number;
+  url?: string;
+  step?: Partial<Step>;
+  error?: Error;
+  root_fields?: Record<string, unknown>;
+  payload?: Record<string, unknown>;
+  blob?: string;
+};
+
+function getMetadata() {
+  return {
+    process: {
+      pid: process.pid,
+      ppid: process.ppid,
+      title: process.title,
+      args: process.argv,
+    },
+    os: {
+      platform: process.platform,
+    },
+    package: {
+      name,
+      version,
+    },
+  };
+}
+
+function formatHTTPVersion(protocol: string) {
+  if (protocol === 'h2') {
+    return 2;
+  } else if (protocol === 'http/1.1') {
+    return 1.1;
+  } else if (protocol === 'http/1.0') {
+    return 1.0;
+  } else if (protocol && protocol.startsWith('h3')) {
+    return 3;
+  }
+  return;
+}
+
+export function formatNetworkFields(network: NetworkInfo) {
+  const { request, response, url } = network;
+  const postdata = request.postData || '';
+  const tls = response?.securityDetails;
+
+  return {
+    // URL and USER AGENT would be parsed and mapped by heartbeat
+    url,
+    user_agent: request.headers?.['User-Agent'],
```

Comment:
APM server uses ES https://www.elastic.co/guide/en/elasticsearch/reference/current/user-agent-processor.html to parse the user agent fields, I would like heartbeat do the same thing if possible when writing our docs.

Reply:
Hmmm, while this is possible heartbeat actually doesn't use ingest node at all at the moment. I'd hate to introduce it for a single field. Additionally, I don't know if there's a huge benefit to this field for synthetics, we could get the browser version in other ways without parsing the UA since we know which browser we're on.

Could we just set `user_agent.original` and `user_agent.version` and `user_agent.name` as fields? That data should be readily available in synthetics and should simplify things greatly.

Reply:
Hmmm that should be a minimal change and we can keep it like that for now until we need the full blown searching and other capability .